src/main/java/seedu/sellsavvy/logic/commands/ordercommands/AddOrderCommand.java

```diff
@@ -1,17 +1,99 @@
+
+import java.util.List;
+
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.commons.util.ToStringBuilder;
+import seedu.sellsavvy.logic.Messages;
 import seedu.sellsavvy.logic.commands.Command;
 import seedu.sellsavvy.logic.commands.CommandResult;
 import seedu.sellsavvy.logic.commands.exceptions.CommandException;
 import seedu.sellsavvy.model.Model;
+import seedu.sellsavvy.model.order.Order;
+import seedu.sellsavvy.model.order.OrderList;
+import seedu.sellsavvy.model.person.Person;
 
 /**
  * Adds an order under a specified person.
  */
 public class AddOrderCommand extends Command {
 
+    public static final String COMMAND_WORD = "addOrder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds an order under the specified person. "
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_ITEM + "ITEM "
+            + PREFIX_DATE + "DELIVERY_BY "
+            + PREFIX_COUNT + "ITEM_COUNT\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_ITEM + "Lamp "
+            + PREFIX_DATE + "20-12-2024 "
+            + PREFIX_COUNT + "2";
+
+    public static final String MESSAGE_SUCCESS = "New Order added for %1$s: %2$s";
+    public static final String MESSAGE_DUPLICATE_WARNING = "Note: "
+            + "This customer already has an order for this item"
+            + ", verify if this is a mistake\n";
+
+    private final Index index;
+    private final Order toAdd;
+
+    /**
+     * Creates an AddOrderCommand to add the specific order under
+     */
+    public AddOrderCommand(Index index, Order order) {
+        requireAllNonNull(index, order);
+        this.index = index;
+        this.toAdd = order;
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return null; // to be implemented
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToAddUnder = lastShownList.get(index.getZeroBased());
+        OrderList orderList = personToAddUnder.getOrderList();
+        String feedbackToUser = orderList.contains(toAdd) ? MESSAGE_DUPLICATE_WARNING : "";
+        orderList.add(toAdd);
+
+        return new CommandResult(feedbackToUser + String.format(MESSAGE_SUCCESS,
+                personToAddUnder.getName(), Messages.format(toAdd)));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        //instanceof handles nulls
+        if (!(other instanceof AddOrderCommand)) {
+            return false;
+        }
+
+        AddOrderCommand otherAddOrderCommand = (AddOrderCommand) other;
+        boolean samePerson = index.equals(otherAddOrderCommand.index);
```

Comment:
actually same index may not necessary means same person as we can addorder then use find to reshuffle the list and then use add order again. Even if index is same, it may not be same person.

Reply:
I'm following the same form of equality as `EditPersonCommand`, where index is one of the attributes that determine whether two of the same commands are equal even if the index may not necessarily refer to the same person.
Commands only care about whether their attributes are the same to determine equality, only on runtime when they are executed then they determine which person the index is referring to on the spot.

Reply:
actually all I meant is to not call the variable name samePerson becuase it is not accurate lol. Additionally, it should be isSameIndex/isSamePerson all boolean should sound like question. EditPersonCommand also never create a variable isSamePerson because it is controversial if it is same person, they just use it directly.
![image](https://github.com/user-attachments/assets/e4cbc6c6-0346-4c2e-9a51-259873f47ce8)

Reply:
sure